src/devices/cpu/sh/sh7021.cpp

```diff
@@ -1,164 +1,2169 @@
+	LOGMASKED(LOG_SCI_WR, "%s:         Receive-Full Interrupt Enable: %d\n", machine().describe_context(), BIT(data, 6));
+	LOGMASKED(LOG_SCI_WR, "%s:         Transmit Enable: %d\n", machine().describe_context(), BIT(data, 5));
+	LOGMASKED(LOG_SCI_WR, "%s:         Receive Enable: %d\n", machine().describe_context(), BIT(data, 4));
+	LOGMASKED(LOG_SCI_WR, "%s:         Multiprocessor Interrupt Enable: %d\n", machine().describe_context(), BIT(data, 3));
+	LOGMASKED(LOG_SCI_WR, "%s:         Transmit-End Interrupt Enable: %d\n", machine().describe_context(), BIT(data, 2));
+	LOGMASKED(LOG_SCI_WR, "%s:         Clock Enable Mode: %d%d\n", machine().describe_context(), BIT(data, 1), BIT(data, 0));
+	m_sci[Channel].scr = data;
+}
+
+template uint8_t sh7021_device::sci_tdr_r<0>();
+template uint8_t sh7021_device::sci_tdr_r<1>();
+
+template <int Channel>
+uint8_t sh7021_device::sci_tdr_r()
+{
+	LOGMASKED(LOG_SCI_RD, "%s: Transmit Data Register %d, sci_tdr_r: %02x\n", machine().describe_context(), Channel, m_sci[Channel].tdr);
+	return m_sci[Channel].tdr;
+}
+
+template void sh7021_device::sci_tdr_w<0>(uint8_t data);
+template void sh7021_device::sci_tdr_w<1>(uint8_t data);
+
+template <int Channel>
+void sh7021_device::sci_tdr_w(uint8_t data)
+{
+	LOGMASKED(LOG_SCI_WR, "%s: Transmit Data Register %d, sci_tdr_w = %02x\n", machine().describe_context(), Channel, data);
+	m_sci[Channel].tdr = data;
+}
+
+template uint8_t sh7021_device::sci_ssr_r<0>();
+template uint8_t sh7021_device::sci_ssr_r<1>();
+
+template <int Channel>
+uint8_t sh7021_device::sci_ssr_r()
+{
+	LOGMASKED(LOG_SCI_RD, "%s: Serial Status Register %d, sci_ssr_r: %02x\n", machine().describe_context(), Channel, m_sci[Channel].ssr);
+	LOGMASKED(LOG_SCI_RD, "%s:         Transmit Empty Flag: %d\n", machine().describe_context(), Channel, BIT(m_sci[Channel].ssr, 7));
+	LOGMASKED(LOG_SCI_RD, "%s:         Receiver Full Flag: %d\n", machine().describe_context(), Channel, BIT(m_sci[Channel].ssr, 6));
+	LOGMASKED(LOG_SCI_RD, "%s:         Overrun Error Flag: %d\n", machine().describe_context(), Channel, BIT(m_sci[Channel].ssr, 5));
+	LOGMASKED(LOG_SCI_RD, "%s:         Framing Error Flag: %d\n", machine().describe_context(), Channel, BIT(m_sci[Channel].ssr, 4));
+	LOGMASKED(LOG_SCI_RD, "%s:         Parity Error Flag: %d\n", machine().describe_context(), Channel, BIT(m_sci[Channel].ssr, 3));
+	LOGMASKED(LOG_SCI_RD, "%s:         Transmit End Flag: %d\n", machine().describe_context(), Channel, BIT(m_sci[Channel].ssr, 2));
+	LOGMASKED(LOG_SCI_RD, "%s:         Multiprocessor Bit Flag: %d\n", machine().describe_context(), Channel, BIT(m_sci[Channel].ssr, 1));
+	LOGMASKED(LOG_SCI_RD, "%s:         Multiprocessor Bit Transfer Flag: %d\n", machine().describe_context(), Channel, BIT(m_sci[Channel].ssr, 0));
+	m_sci[Channel].ssr_read |= m_sci[Channel].ssr & 0xf8;
+	return (m_sci[Channel].ssr & 0xf9) | 0x04;
```

Comment:
You should guard against debugger accesses in read handlers (`machine().side_effects_disabled()`).